File: 백준/Silver/10845. 큐/큐.py
import sys
import queue


input = sys.stdin.readline

n=input()
qu = []

for i in range(int(n)):
    m = input().split()
    if m[0] ==  "push" : 
       qu.append(m[1])
    elif m[0] == "pop" : 
        if qu : 
            print(qu.pop(0))
        else : 
            print(-1)
    elif m[0] == 'size' : 
        print(len(qu))

    elif m[0] == 'empty' :
        if len(qu) == 0 : 
            print(1)
        else : 
            print(0)

    elif m[0] == 'front' :
        if len(qu) == 0 : 
            print(-1)
        else : 
            print(qu[0])
    
    elif m[0] == 'back' :
        if len(qu) == 0 : 
            print(-1)
        else : 
            print(qu[-1])

File: 백준/Silver/10845. 큐/큐.cc
#include <iostream>
#include <queue>

using namespace std;

int main()
{
    int n =0;
    cin >> n;
    queue<int> que; // 큐

    for (int i = 0; i < n; i++)
    {
        string str; // 큐 행동을 처리할 문자열
        cin >> str; // 문자열 입력
        if (str == "push") 
        {
            int tep;
            cin >> tep; // push일 경우에만 추가 입력
            que.push(tep);
        }
        else if (str == "front")
        {
            if (que.empty()) cout << -1 << endl;
            else
            {
                cout << que.front() << endl;// 큐 front 출력
            }
        }
        else if (str == "back")
        {
            if (que.empty()) cout << -1 << endl;
            else
            {
                cout << que.back() << endl;// 큐 back 출력
            }
        }
        else if (str == "empty")
        {
            if (que.empty()) cout << 1 << endl;
            else cout << 0 << endl;
        }
        else if (str == "pop")
        {
            if (que.empty())
                cout << -1 << endl;
            else
            {
                cout << que.front() << endl;
                que.pop();
            }
            
        }
        else if (str == "size")
        {
            cout << que.size() << endl;
        }
    }
}